### ISSUE TITLE
Handle path argument of URL.build, which doesn't start from /

### ISSUE THREAD

**Issue by rominf** [aio-libs/yarl#242]
I'm building `URL` with this line of code:
```
url = URL(scheme='https', host='profile-srv.kama.gs', path='oauth/token')
```

Then I pass `url` to `session.post` and get 400 error.

When I construct it with:
```
url = URL(scheme='https', host='profile-srv.kama.gs', path='/oauth/token')
```
or convert it to `str`:
```
url = str(URL(scheme='https', host='profile-srv.kama.gs', path='/oauth/token'))
```
everything is OK.

I suggest prepending `/` if `path` doesn't start with `/` or, at least, raise an exception.

**Comment by asvetlov** [comment]
I think the rule should be: URLs with authority (host part not empty) should raise ValueError on processing relative path (not started with `/`).

Would you provide a patch?

**Comment by rominf** [comment]
@asvetlov, OK, I will, wait.

**Comment by rominf** [comment]
@asvetlov, see #243

**Comment by asvetlov** [comment]
Fixed by #243
